bin/brew

```diff
@@ -23,7 +23,7 @@ unset GEM_HOME
 unset GEM_PATH
 
 BREW_SYSTEM=$(uname -s | tr "[:upper:]" "[:lower:]")
-if [ "$BREW_SYSTEM" = "darwin" ]
+if [ "$BREW_SYSTEM" = "darwin" ] && [ -z "$HOMEBREW_OLD_RUBY_TEST" ]
```

Comment:
I'd rather not have a general approach; I don't want to encourage this behaviour.